### PR TITLE
87 interface for schedule t

### PR DESCRIPTION
**Pull request by SolviQorda**
Hi @turion - as suggested :)

### CODE REVIEW

File: rhine/src/Control/Monad/Schedule.hs [hunk 1]
```diff
@@ -1,13 +1,17 @@
 {- |
-This module supplies a general purpose monad transformer
-that adds a syntactical "delay", or "waiting" side effect.
+    This module supplies a general purpose monad transformer
```
Comment (turion) [inline rhine/src/Control/Monad/Schedule.hs]: Can you unindent this?

File: rhine/src/Control/Monad/Schedule.hs [hunk 2]
```diff
@@ -1,13 +1,17 @@
...
-This allows for universal and deterministic scheduling of clocks
-that implement their waiting actions in 'ScheduleT'.
-See 'FRP.Rhine.Schedule.Trans' for more details.
+    This allows for universal and deterministic scheduling of clocks
+    that implement their waiting actions in 'ScheduleT'.
+    See 'FRP.Rhine.Schedule.Trans' for more details.
 -}
 
-{-# LANGUAGE DeriveFunctor #-}
+{-# LANGUAGE DeriveFunctor         #-}
```
Comment (turion) [inline rhine/src/Control/Monad/Schedule.hs]: Too many spaces ;)

File: rhine/src/Control/Monad/Schedule.hs [hunk 3]
```diff
@@ -37,31 +41,25 @@ Values in @ScheduleT diff m@ are delayed computations with side effects in 'm'.
 Delays can occur between any two side effects, with lengths specified by a 'diff' value.
 These delays don't have any semantics, it can be given to them with 'runScheduleT'.
 -}
+
 type ScheduleT diff = FreeT (Wait diff)
 
+--  Implement type class MonadWait with method wait
+class Monad m => MonadWait diff m where
+    wait :: diff -> m ()
```
Comment (turion) [inline rhine/src/Control/Monad/Schedule.hs]: Standard indentation is 2 spaces in Haskell

File: rhine/src/Control/Monad/Schedule.hs [hunk 4]
```diff
@@ -37,31 +41,25 @@ Values in @ScheduleT diff m@ are delayed computations with side effects in 'm'.
 Delays can occur between any two side effects, with lengths specified by a 'diff' value.
 These delays don't have any semantics, it can be given to them with 'runScheduleT'.
 -}
+
```
Comment (turion) [inline rhine/src/Control/Monad/Schedule.hs]: Superfluous newline

File: rhine/src/Control/Monad/Schedule.hs [hunk 5]
```diff
@@ -37,31 +41,25 @@ Values in @ScheduleT diff m@ are delayed computations with side effects in 'm'.
...
 
--- | Run a 'ScheduleT' value in a 'MonadIO',
---   interpreting the times as milliseconds.
-runScheduleIO
-  :: (MonadIO m, Integral n)
-  => ScheduleT n m a -> m a
-runScheduleIO = runScheduleT $ liftIO . threadDelay . (* 1000) . fromIntegral
-
--- TODO The definition and type signature are both a mouthful. Is there a simpler concept?
+--TODO The definition and type signature are both a mouthful. Is there a simpler concept?
```
Comment (turion) [inline rhine/src/Control/Monad/Schedule.hs]: Don't need to touch this line

**Comment by SolviQorda** [comment]
Hi @turion - thanks for the git advice, I'll use those in future (and fix the superfluous spacing :see_no_evil: )

In terms of the review - I hear what you're saying, and I'll try and take the comments you've given and try and implement MonadSchedule, then request a review.

Thanks again!

**Comment by turion** [comment]
Superseded by https://github.com/turion/rhine/pull/171

**Pull request closed without merging**